### ISSUE TITLE
Don't count form feed character (`\f` or `^L`) as newline, optionally

### ISSUE THREAD

**Issue by skangas** [codespell-project/codespell#2609]
In Emacs, the [form feed](https://en.wikipedia.org/wiki/Page_break#Form_feed) `\f` or `^L` (Ctrl+L) character, 0xC in ASCII, is not considered to start a new line. Wikipedia explains:

> The form feed character is sometimes used in plain text files of source code as a delimiter for a page break, or as marker for sections of code. Some editors, in particular [emacs](https://en.wikipedia.org/wiki/Emacs) and [vi](https://en.wikipedia.org/wiki/Vi), have built-in commands to page up/down on the form feed character. This convention is predominantly used in [Lisp](https://en.wikipedia.org/wiki/Lisp_(programming_language)) code, and is also seen in [C](https://en.wikipedia.org/wiki/C_(programming_language)) and [Python](https://en.wikipedia.org/wiki/Python_(programming_language)) source code. [GNU](https://en.wikipedia.org/wiki/GNU) Coding Standards require such form feeds in C.[[2]](https://en.wikipedia.org/wiki/Page_break#cite_note-2) Editors like [Vim](https://en.wikipedia.org/wiki/Vim_(text_editor)) and [Emacs](https://en.wikipedia.org/wiki/Emacs) understand such sections and have shortcuts for moving among them.

In Emacs, a file containing just the three characters `\n\f\d` will be considered to have _two_ (2) lines, the first of them being `\n` and the second line `\f\n`. But codespell counts this file as a file with _three_ (3) lines.

Recipe to reproduce:

```
$ echo -ne "foo\n\f\nte\n" > /tmp/foo.txt
$ codespell /tmp/foo.txt
/tmp/foo.txt:4: te ==> the, be, we, to
```

If I open `/tmp/foo.txt` in Emacs, and try to jump to line 4, I end up on the empty line at the end of the file. I do not go to the line containing the typo, which in Emacs is line 3. This obviously gets worse the more `\n\f\n` there are in a file: every one means we land further and further from the actual typo.

Would it be possible to add an option to treat `\f` in the way that is expected by Emacs? I'm not sure what it should be called, but something like `--form-feed-no-newline` or `--emacs-form-feed` perhaps.

This would help tremendously when using codespell on Emacs Lisp source code. There are many, many Lisp files which contain the character sequence `\n\f\n`.

Thanks.

**Comment by DimitriPapadopoulos** [comment]
It looks `more` understands this too, by default:
```
$ cat foo.txt 
foo


te
$ 
$  more  foo.txt 
foo
^L
te
$ 
```
Gnome editors too, by default as well:
![Geany](https://user-images.githubusercontent.com/3234522/202844622-f829b56c-b98f-404d-bd14-412566bb502d.png)

Perhaps it should be the default in codespell too.

**Comment by skangas** [comment]
Yes, indeed it could/should be the default.

However, when I re-run the above using codespell from the master branch I do get the expected result:

```
/tmp/foo.txt:3: te ==> the, be, we, to
```

This is surprising to me, as the version I tried with previously was the latest released 2.2.2 on Debian. I can't see any relevant changes in codespell itself in the intervening period. Perhaps some bug was fixed in a dependency that was not yet updated in Debian?

Are you seeing the same as me when running current `master`?
